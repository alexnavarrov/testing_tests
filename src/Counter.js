import React, {useState} from 'react';

function Counter({initialNumber}) {
    const [count, setCount] = useState(initialNumber);

    return (
        <div>
            <p>Holi</p>
            <p data-testid="value">{count}</p>
            <button data-testid="button" onClick={() => setCount(count + 1)}>
                Click me!
            </button>
        </div>
    );
}

export default Counter;