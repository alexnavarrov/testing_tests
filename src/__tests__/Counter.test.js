import React from "react";
import { render, screen, fireEvent } from '@testing-library/react';
import Counter from "../Counter";

describe('Components | Counter', () => {
    
    it('renders the component', () => {
        const props = {initialNumber: 2};
        const { container } = render(<Counter {...props} />);
        expect(container).toMatchSnapshot();
    });


    it('increments the counter', () => {
        const props = {initialNumber: 4};
        const { rerender } = render(<Counter {...props} />);
        const value = screen.getByTestId('value');
        //const button = screen.getByTestId('button');
        expect(value.textContent).toBe('4');

    })
});