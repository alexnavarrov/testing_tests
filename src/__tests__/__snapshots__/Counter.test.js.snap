// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | Counter renders the component 1`] = `
<div>
  <div>
    <p>
      Holi
    </p>
    <p
      data-testid="value"
    >
      2
    </p>
    <button
      data-testid="button"
    >
      Click me!
    </button>
  </div>
</div>
`;
